scene/resources/curve.h

```diff
@@ -238,24 +238,37 @@ class Curve3D : public Resource {
 
 	Vector<Point> points;
 
-	struct BakedPoint {
-		real_t ofs = 0.0;
-		Vector3 point;
-	};
+	// unused, remove?
+	//struct BakedPoint {
+	//	real_t ofs = 0.0;
+	//	Vector3 point;
+	//};
 
 	mutable bool baked_cache_dirty = false;
 	mutable PackedVector3Array baked_point_cache;
 	mutable Vector<real_t> baked_tilt_cache;
+	// cache the rotation
 	mutable PackedVector3Array baked_up_vector_cache;
+
 	mutable Vector<real_t> baked_dist_cache;
 	mutable real_t baked_max_ofs = 0.0;
 
 	void mark_dirty();
 
 	void _bake() const;
 
+	struct Interval {
+		int idx;
+		real_t frac;
+	};
+	Interval _find_interval(real_t p_offset) const;
+	Vector3 _sample_baked(Interval p_interval, bool p_cubic) const;
+	real_t _sample_baked_tilt(Interval p_interval) const;
+	Basis _sample_baked_basis(Interval p_interval, bool p_apply_tilt = false) const;
+
 	real_t bake_interval = 0.2;
 	bool up_vector_enabled = true;
+	bool loop_smoothing_enabled = true;
```

Comment:
As long as `Curve3D` is a closed path, I can't imagine a use case where this is `false`. If we want it to be `false`, we can simply make it an open path with overlapping vertices, like Illustrator's scissors tool. So I think you can remove this and always enable smoothing for closed paths.

Reply:
> I can't imagine a use case where this is false

Me neither. This option is left just in case. Will remove it in new commit.